### DISCUSSION TITLE
How to add a custom build artifact to the specific project type or configuration?

### DISCUSSION THREAD

**Discussion by baoruiqi** [eclipse-embed-cdt/eclipse-plugins#564] [Managed Build Plug-ins Q&A]
Hi, all

I'm trying to integrate Embedded CDT into my project and add some functionality.

I have read the document of [Managed Build System Extensibility Document](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.cdt.doc.isv%2Fguide%2Fmbs%2FextensibilityGuide%2FManaged_Build_Extensibility.html) and know how to define a new project type and configure related properties to be integrated into Eclipse IDE's Managed build system as a plug-in, but this is not enough to achieve my purpose.

![image](https://github.com/eclipse-embed-cdt/eclipse-plugins/assets/128341204/631d4e2e-d37f-46ba-bf98-a179795cb963)

In fact, I hope to add a custom build artifact to the RISC-V Cross Toolchain of the existing c/c++ managed project, so that the project can be built as an unlinked object file, such as .o format. That is to say, I only need to use the compiler in the toolchain tool for the time being
. However, when switching to other predefined artifacts such as executable or static library, it can still work normally.

In order to achieve this goal, do I need to modify the source code of Embedded CDT? By the way, if I want to create a new unlinked object type based on the buildDefinition mechanism, how can I inherit the settings of the RISC-V Toolchain in Embedded CDT and further achieve my goal.

@ilg-ul @jonahgraham @ruspl-afed

**Comment by baoruiqi** [comment]
I think the most ideal way is to use CDT's tool integrator mechanism to create a new Project type and inherit all relevant configurations of the RISC-V toolchain in Embedded CDT, and additionally define a new build artifact to execute compiling without linking operation.

In this case, there is no need to modify the source code of Embedded CDT. However, I don't know how to implement it and find it hard to locate the operations in source code corresponding to the new build artifact.

**Reply by jonahgraham** [reply]
Sorry that I don't have further info on this.
